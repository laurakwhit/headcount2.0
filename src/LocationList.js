import React from 'react';

const LocationList = ({ locations, selectLocation }) => {
    const locationButtons = locations.map((location, i) => (
        <button key={i}
            name={location}
            className='location-btn'
            onClick={() => selectLocation(location)} >
            {location}
        </button>
    ))

    return (
        <div>
            <aside>
                {locationButtons}
            </aside>
        </div>
    )
}

export default LocationList;