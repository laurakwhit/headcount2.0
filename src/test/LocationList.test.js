import React from 'react';
import LocationList from '../LocationList';
import { shallow } from 'enzyme';

describe('LocationList', () => {
    let wrapper;
    let selectLocationMock;

    beforeEach(() => {
        selectLocationMock = jest.fn()
        wrapper = shallow(<LocationList locations={['Colorado']} selectLocation={selectLocationMock} />)
    })

    it('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should call selectLocation when location button is clicked', () => {
        wrapper.find('.location-btn').simulate('click')

        expect(selectLocationMock).toHaveBeenCalled()
    })
})