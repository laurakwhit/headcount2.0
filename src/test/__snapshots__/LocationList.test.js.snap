// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationList should match the snapshot 1`] = `"<div><aside><button name=\\"Colorado\\" class=\\"location-btn\\">Colorado</button></aside></div>"`;
